cumulus/client/consensus/aura/src/collators/slot_based/block_builder_task.rs

```diff
@@ -0,0 +1,492 @@
+
+		// We mainly call this to inform users at genesis if there is a mismatch with the
+		// on-chain data.
+		collator.collator_service().check_block_status(parent_hash, &parent_header);
+
+		let slot_claim = match crate::collators::can_build_upon::<_, _, P>(
+			para_slot.slot,
+			para_slot.timestamp,
+			parent_hash,
+			included_block,
+			&*para_client,
+			&keystore,
+		)
+		.await
+		{
+			Some(slot) => slot,
+			None => {
+				tracing::debug!(
+					target: crate::LOG_TARGET,
+					?core_index,
+					slot_info = ?para_slot,
+					unincluded_segment_len = parent.depth,
+					relay_parent = %relay_parent,
+					included = %included_block,
+					parent = %parent_hash,
+					"Not building block."
+				);
+				continue
+			},
+		};
+
+		tracing::debug!(
+			target: crate::LOG_TARGET,
+			?core_index,
+			slot_info = ?para_slot,
+			unincluded_segment_len = parent.depth,
+			relay_parent = %relay_parent,
+			included = %included_block,
+			parent = %parent_hash,
+			"Building block."
+		);
+
+		let validation_data = PersistedValidationData {
+			parent_head: parent_header.encode().into(),
+			relay_parent_number: *relay_parent_header.number(),
+			relay_parent_storage_root: *relay_parent_header.state_root(),
+			max_pov_size,
+		};
+
+		let (parachain_inherent_data, other_inherent_data) = match collator
+			.create_inherent_data(
+				relay_parent,
+				&validation_data,
+				parent_hash,
+				slot_claim.timestamp(),
+			)
+			.await
+		{
+			Err(err) => {
+				tracing::error!(target: crate::LOG_TARGET, ?err);
+				break
+			},
+			Ok(x) => x,
+		};
+
+		let validation_code_hash = match code_hash_provider.code_hash_at(parent_hash) {
+			None => {
+				tracing::error!(target: crate::LOG_TARGET, ?parent_hash, "Could not fetch validation code hash");
+				break
+			},
+			Some(v) => v,
+		};
+
+		check_validation_code_or_log(&validation_code_hash, para_id, &relay_client, relay_parent)
+			.await;
+
+		let Ok(Some(candidate)) = collator
+			.build_block_and_import(
+				&parent_header,
+				&slot_claim,
+				None,
+				(parachain_inherent_data, other_inherent_data),
+				authoring_duration,
+				// Set the block limit to 50% of the maximum PoV size.
+				//
+				// TODO: If we got benchmarking that includes the proof size,
+				// we should be able to use the maximum pov size.
+				(validation_data.max_pov_size / 2) as usize,
+			)
+			.await
+		else {
+			tracing::error!(target: crate::LOG_TARGET, "Unable to build block at slot.");
+			continue;
+		};
+
+		let new_block_hash = candidate.block.header().hash();
+
+		// Announce the newly built block to our peers.
+		collator.collator_service().announce_block(new_block_hash, None);
+
+		if let Err(err) = collator_sender.unbounded_send(CollatorMessage {
+			relay_parent,
+			parent_header,
+			parachain_candidate: candidate,
+			validation_code_hash,
+			core_index: *core_index,
+		}) {
+			tracing::error!(target: crate::LOG_TARGET, ?err, "Unable to send block to collation task.");
```

Comment:
```suggestion
			tracing::error!(target: crate::LOG_TARGET, ?err, "Unable to send block to collation task.");
			break
```

This means that the collation task is dead and thus, we should also end the block production.